src/services/ogc-features/src/main/java/com/camptocamp/opendata/ogc/features/model/GeoToolsFeatureCollection.java

```diff
@@ -49,9 +52,11 @@ public Stream<GeodataRecord> getFeatures() {
         int characteristics = Spliterator.DISTINCT | Spliterator.NONNULL;
         Spliterator<SimpleFeature> spliterator = Spliterators.spliteratorUnknownSize(iterator, characteristics);
         Stream<SimpleFeature> stream = StreamSupport.stream(spliterator, false);
-        stream = stream.onClose(() -> {
-            it.close();
-        });
-        return stream.map(featureMapper);
+        stream = stream.onClose(it::close);
+        if (!StringUtils.isEmpty(targetCrs)) {
+            return gtProcessors.reproject(String.format("EPSG:%s", targetCrs)).apply(stream.map(featureMapper));
+        } else {
+            return gtProcessors.reproject("EPSG:4326").apply(stream.map(featureMapper));
+        }
```

Comment:
Wondering if there is a gain to check if the featureset requires reprojection and avoid it if not (in case the datasource is expressed in the same projection as the one requested).

2 cases here, where the reprojection is not necessary:
* target CRS is provided as argument, but the featureset is already in the requested CRS
* target CRS is not provided, but the featureset is in EPSG:4326.